src/services/RPCSelector.ts

```diff
@@ -0,0 +1,79 @@
+import { providers } from 'ethers';
+import { isTimestampMoreRecentThan } from '../utils/helpers';
+
+interface ProviderComparand {
+  blockNumber: number;
+  url: string;
+}
+
+interface Config {
+  rpcRequestTimeout: number;
+  maxBlockAge: number;
+}
+
+class RPCSelector {
+  private readonly urls: string[];
+  private readonly rpcRequestTimeout: number;
+  private readonly maxBlockAge: number;
+
+  constructor(urls: string | string[], config: Config = { rpcRequestTimeout: 15000, maxBlockAge: 60 }) {
+    this.urls = typeof urls === 'string' ? urls.split(',') : urls;
+    this.rpcRequestTimeout = config.rpcRequestTimeout;
+    this.maxBlockAge = config.maxBlockAge;
+  }
+
+  public async selectByTimestamp(): Promise<string> {
```

Comment:
Can we return Provider right away?

Reply:
I've decided to keep it like this because the user might want to instantiate a different provider other than `JsonRpc` or `WS`
I also had a few problems by returning the instance of the provider.

Reply:
That's right Provider interface encapsulates the details.